src/Models/UserModel.php

```diff
@@ -185,48 +184,51 @@ public function activate(User $user): void
     {
         $user->active = true;
 
-        $return = $this->save($user);
-
-        $this->checkQueryReturn($return);
+        $this->saveWithEmailIdentity($user);
     }
 
     /**
-     * Override the BaseModel's `save()` method to allow
-     * updating of user email, password, or password_hash
-     * fields if they've been modified.
-     *
-     * @param array|User $data
-     *
-     * @TODO can't change the return type to void.
+     * Save User and its Email Identity (email, password, or password_hash fields)
+     * if they've been modified.
      */
-    public function save($data): bool
+    public function saveWithEmailIdentity(User $data): void
```

Comment:
Did we need to rename this for a specific reason? If so, or the team majority says we should then ok, but I'm in favor of keeping it as an override of `Model::save`. 

I understand this is a more specific naming, but the whole purpose of the override was to make it as seamless as possible for the developer so they didn't have to think too much about it. This forces them to remember a new method for that single model, which is inconvenient and, as far as I can tell, unnecessary.

Reply:
Reading back through I see the discussion between you two so I'm overruled here. Just going on the record to say I disagree :)

Reply:
👹 Contention noted!

I think this does two good things for us:
1. It lets us still use `save()` without messing with identities.
2. It allows us to force the `User` type parameter which makes this method much easier to handle. 

Also just to say it again: this should just be a temporary measure until we get better Model event parameters in place.

Reply:
I overrode `save()` again.

After all, if save() is going to be used in the future, it is better for users to use save() now.

It is confusing only save() can save Email Identity, but users do not need to use insert() or update().

Reply:
Alternative idea...

The whole reason we override this method is that `afterInsert` and `afterUpdate` receive the data as an array, not the original `User` entity, and that array is filtered by `$allowedFields` so the credentials are stripped. What if we override `insert()` and `update()` to store credentials in a new private property before calling `parent::method()` and then still use the `after____` methods to handle the identities, just using the stored versions instead?

Reply:
Added `UserModel::saveEmailIdentity()` and the Model Events call it when afterInsert and afterUpdate.